### ISSUE TITLE
what's a reasonable loss for blip-2 stage2 training?

### ISSUE THREAD

**Issue by ldfandian** [salesforce/LAVIS#463]
Thanks for the amazing works you guys make.

I am just begining to train on a single A100 80G (with batch_size=35, accum_grad_iters=50).
Effectively, I guess real loss is around 0.05*50=2.5. 
Can you please shed some light on what's the reasonably good converged loss ?

![image](https://github.com/salesforce/LAVIS/assets/5018331/2ec7c227-1369-4ead-9348-d1be72e769c3)

**Comment by ldfandian** [comment]
@LiJunnan1992 can you please give some advice?

I am trying to pretrain blip-2 from a scratch. My change is to connect vit-b-16 to vicuna-7b~
I used the dataset of vg/coco/cc3m/cc12m/sbu (~10m in total), the model loss did converge very slowly on both stage1 and stage2.

BUT, when I try to do generate on the model, the output sentense has nothing to do with the image content at all (although the sentence itself is very readable). Even that I use an image in the training set, it is the same thing...

Anything wrong here?
- is dataset too small?
- is the loss not converged good enough?
- or anything wrong with my code?

== stage 1 ==
`
    "run": {
        "task": "image_text_pretrain",
        "runner": "runner_iter",
        "max_iters": 200000.0,
        "iters_per_inner_epoch": 5000.0,
        "train_dataset_ratios": {
            "cc3m_cc12m_sbu_caption": 0.6,
            "coco_caption": 0.03,
            "vg_caption": 0.03
        },
        "lr_sched": "linear_warmup_step_lr",
        "init_lr": 0.0001,
        "min_lr": 1e-05,
        "warmup_lr": 1e-06,
        "lr_decay_rate": 0.93,
        "weight_decay": 0.05,
        "max_epoch": 20,
        "batch_size_train": 380,
        "accum_grad_iters": 4,
        "num_workers": 4,
        "warmup_steps": 5000,
        ... ...
`

I stops the training at checkpoint_135000 (which means 170k steps * 380 images/step=~64.6m images (around 6x total image count)
`
...
{"train_lr": "0.000", "train_loss": "1.641", "train_loss_itc": "2.437", "train_loss_itm": "0.513", "train_loss_lm": "3.614"}
{"train_lr": "0.000", "train_loss": "1.635", "train_loss_itc": "2.417", "train_loss_itm": "0.510", "train_loss_lm": "3.612"}
{"train_lr": "0.000", "train_loss": "1.631", "train_loss_itc": "2.414", "train_loss_itm": "0.510", "train_loss_lm": "3.601"}
{"train_lr": "0.000", "train_loss": "1.630", "train_loss_itc": "2.400", "train_loss_itm": "0.508", "train_loss_lm": "3.613"}
{"train_lr": "0.000", "train_loss": "1.628", "train_loss_itc": "2.392", "train_loss_itm": "0.507", "train_loss_lm": "3.613"}
{"train_lr": "0.000", "train_loss": "1.626", "train_loss_itc": "2.383", "train_loss_itm": "0.505", "train_loss_lm": "3.615"}
{"train_lr": "0.000", "train_loss": "1.624", "train_loss_itc": "2.382", "train_loss_itm": "0.505", "train_loss_lm": "3.608"}
`

== stage 2 ==

`
{
    "run": {
        "task": "image_text_pretrain",
        "runner": "runner_iter",
        "max_iters": 400000.0,
        "iters_per_inner_epoch": 10000.0,
        "train_dataset_ratios": {
            "cc3m_cc12m_sbu_caption": 0.6,
            "coco_caption": 0.02,
            "vg_caption": 0.02
        },
        "lr_sched": "linear_warmup_step_lr",
        "init_lr": 0.0001,
        "min_lr": 5e-05,
        "warmup_lr": 1e-06,
        "lr_decay_rate": 0.95,
        "weight_decay": 0.05,
        "max_epoch": 40,
        "batch_size_train": 20,
        "accum_grad_iters": 20,
        "batch_size_eval": 64,
        "num_workers": 16,
        "warmup_steps": 10000,
        ... ...
`

the training is now at checkpoint_300000(which means 300k steps * 20 images per step per gpu * 4 gpus=~24m images (around 2x total image count)

`
... ...
{"train_lr": "0.000", "train_loss": "0.111"}
{"train_lr": "0.000", "train_loss": "0.111"}
{"train_lr": "0.000", "train_loss": "0.110"}
{"train_lr": "0.000", "train_loss": "0.110"}
{"train_lr": "0.000", "train_loss": "0.110"}
{"train_lr": "0.000", "train_loss": "0.110"}
{"train_lr": "0.000", "train_loss": "0.109"}
{"train_lr": "0.000", "train_loss": "0.110"}
`

**Comment by xingenju** [comment]
What's your log for Flan-T5?
